xray/plot/facetgrid.py

```diff
@@ -0,0 +1,384 @@
+            Font size as used by matplotlib text
+        kwargs :
+            additional keyword arguments to plotfunc
+
+        Returns
+        -------
+        self : FacetGrid object
+
+        '''
+        import matplotlib.pyplot as plt
+
+        # These should be consistent with xray.plot._plot2d
+        cmap_kwargs = {'plot_data': self.darray.values,
+                'vmin': None,
+                'vmax': None,
+                'cmap': None,
+                'center': None,
+                'robust': False,
+                'extend': None,
+                'levels': 7 if 'contour' in plotfunc.__name__ else None, # MPL default
+                'filled': plotfunc.__name__ != 'contour',
+                }
+
+        # Allow kwargs to override these defaults
+        for param in kwargs:
+            if param in cmap_kwargs:
+                cmap_kwargs[param] = kwargs[param]
+
+        # colormap inference has to happen here since all the data in self.darray
+        # is required to make the right choice
+        cmap_params = _determine_cmap_params(**cmap_kwargs)
+
+        if 'contour' in plotfunc.__name__:
+            # extend is a keyword argument only for contour and contourf, but
+            # passing it to the colorbar is sufficient for imshow and
+            # pcolormesh
+            kwargs['extend'] = cmap_params['extend']
+            kwargs['levels'] = cmap_params['levels']
+
+        defaults = {
+                'add_colorbar': False,
+                'add_labels': False,
+                'norm': cmap_params.pop('cnorm'),
+                }
+
+        # Order is important
+        defaults.update(cmap_params)
+        defaults.update(kwargs)
+
+        for d, ax in zip(self.name_dicts.flat, self.axes.flat):
+            # None is the sentinel value
+            if d is not None:
+                subset = self.darray.loc[d]
+                mappable = plotfunc(subset, x, y, ax=ax, **defaults)
+
+        self.x = x
+        self.y = y
+
+        # Left side labels
+        for ax in self.axes[:, 0]:
+            ax.set_ylabel(self.y)
+
+        # Bottom labels
+        for ax in self.axes[-1, :]:
+            ax.set_xlabel(self.x)
+
+        self.fig.tight_layout()
+
+        if self._single_group:
+            for d, ax in zip(self.name_dicts.flat, self.axes.flat):
+                if d is None:
+                    ax.set_visible(False)
+
+        # colorbar
+        if kwargs.get('add_colorbar', True):
+            cbar = self.fig.colorbar(mappable,
+                                     ax=list(self.axes.flat),
+                                     extend=cmap_params['extend'])
+
+            if self.darray.name:
+                cbar.set_label(self.darray.name, rotation=270,
+                        verticalalignment='bottom')
+
+        # This happens here rather than __init__ since FacetGrid.map should
+        # use default ticks
+        self.set_ticks(max_xticks, max_yticks, fontsize)
```

Comment:
One option is to keep the method and to use it in the docs but not call it by default. Or simply adjusting size might be enough.

On Fri, Sep 4, 2015 at 9:21 AM, Clark Fitzgerald <notifications@github.com>
wrote:

>> +                if d is None:
>> +                    ax.set_visible(False)
>> +
>> +        # colorbar
>> +        if kwargs.get('add_colorbar', True):
>> +            cbar = self.fig.colorbar(mappable,
>> +                                     ax=list(self.axes.flat),
>> +                                     extend=cmap_params['extend'])
>> +
>> +            if self.darray.name:
>> +                cbar.set_label(self.darray.name, rotation=270,
>> +                        verticalalignment='bottom')
>> +
>> +        # This happens here rather than __init__ since FacetGrid.map should
>> +        # use default ticks
>> +        self.set_ticks(max_xticks, max_yticks, fontsize)
> I agree. This was added since the docs look bad. I'll take another look at what Seaborn does and see if we can somehow get away without this.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/xray/xray/pull/527/files#r38768382